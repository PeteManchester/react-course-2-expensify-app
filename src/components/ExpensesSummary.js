import React from "react";
import { connect } from "react-redux";
import numeral from "numeral";
import "numeral/locales/en-gb";
import selectExpensesTotal from "../selectors/expense-total";
import selectExpenses from "../selectors/expenses";

numeral.locale("en-gb");

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
  const expenseWord = expenseCount === 1 ? "expense" : "expenses";
  const formattedExpenseTotal = numeral(expenseTotal / 100).format("$0,0.00");
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal}
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expenseTotal: selectExpensesTotal(visibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
