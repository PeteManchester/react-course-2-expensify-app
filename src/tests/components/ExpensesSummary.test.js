import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render expense summary with one expense', () =>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={1233}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render expense summary with expenses', () =>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={12334462564563562}/>);
    expect(wrapper).toMatchSnapshot();
});


