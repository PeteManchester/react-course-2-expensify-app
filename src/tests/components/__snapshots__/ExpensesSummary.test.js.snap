// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary with expenses 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    £123,344,625,645,635.63
  </h1>
</div>
`;

exports[`should render expense summary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    £12.33
  </h1>
</div>
`;
